starter-code/scripts/controllers/articleController.js

```diff
@@ -14,6 +14,7 @@
   };
 
   // COMMENT: What does this method do?  What is it's execution path?
+  //This method handles the functionality when a user clicks on the 'read on' link of the home page. Once clicked, the function loops through the article array and matches the id that was selected. Finally, the articleController path goes on to call a the second function that renders the full article to the screen.
```

Comment:
Good job! It doesn't really loop through the article array, though. This method is triggered as the first callback upon the client hitting the '/article/:id' route (when a user clicks on "read on"). This captures the id of the article associated with that Read On click, and that id now lives on the ctx object as ctx.params.id.  The method calls Article.findWhere, passing it the 'id' field, the value of that id (ctx.params.id), and a callback function.  Article.findWhere executes a database query that finds the article with that ID in the database, and then, Article.findWhere calls the callback function (articleData).  articleData attaches the article returned from the database query to the ctx object (to reference later) and calls the next method in the route (articleController.index).